docs/application/native/guides/text-input/capi-ui-gesture.md

```diff
@@ -0,0 +1,247 @@
+   The candidates of the gesture types are defined in the `gesture_common.h` header file as the enumerations of `hand_gesture_type_e`:
+
+    ```c
+    void
+    is_supported_gesture_type(hand_gesture_h gesture_h, hand_gesture_type_e gesture_type, bool* is_supported)
+    {
+        int ret;
+        ret = hand_gesture_is_supported_type(gesture_h, gesture_type, is_supported);
+        if (HAND_GESTURE_ERROR_NONE != ret)
+            /* Error handling */
+    }
+    ```
+
+2. Get the information of the Gesture engine service used in current.
+
+   You can get the application ID and name of the current Gesture engine service using `hand_gesture_get_engine_info()`:
+
+    ```c
+    void
+    get_info(hand_gesture_h gesture_h)
+    {
+        int ret;
+        char* engine_app_id;
+        char* engine_name;
+
+        ret = hand_gesture_get_engine_info(gesture_h, &engine_app_id, &engine_name);
+        if (HAND_GESTURE_ERROR_NONE != ret)
+            /* Error handling */
+    }
+    ```
+
+
+<a name="set_option"></a>
+## Set option
+
+You can select an option to detect hand gestures continuously or not. If you want to make the application detect hand gestures continuously, set the option as `HAND_GESTURE_OPTION_ALWAYS_ON` using `hand_gesture_set_option()`:
+
+```c
+void
+set_option(hand_gesture_h gesture_h)
+{
+    int ret;
+    ret = hand_gesture_set_option(gesture_h, HAND_GESTURE_OPTION_ALWAYS_ON);
+    if (HAND_GESTURE_ERROR_NONE != ret)
+        /* Error handling */
+}
+```
+
+
+<a name="callback"></a>
+## Set and unset callbacks
+
+The Gesture provides two types of callbacks for getting error and gesture recognition result.
+The enum values, as well as the parameter details, for the callback parameters are defined in the `gesture.h` and `gesture_common.h` header files.
+
+1. Error
+
+   When an error occurs, the Gesture library sends a message using a callback:
+
+   > [!NOTE]
+   > Ensure that you set the error callback function before calling `hand_gesture_start_recognition()`.
+
+    ```c
+    /* Callback */
+    void
+    error_cb(hand_gesture_h gesture_h, hand_gesture_error_e reason, const char* msg, void* user_data)
+    {
+        /* Your code */
+    }
+
+    /* Set */
+    void
+    set_error_cb(hand_gesture_h gesture_h)
+    {
+        int ret;
+        ret = hand_gesture_set_error_cb(gesture_h, error_cb, NULL);
+        if (HAND_GESTURE_ERROR_NONE != ret)
+            /* Error handling */
+    }
+
+    /* Unset */
+    void
+    unset_error_cb(hand_gesture_h gesture_h)
+    {
+        int ret;
+        ret = hand_gesture_unset_error_cb(gesture_h);
+        if (HAND_GESTURE_ERROR_NONE != ret)
+            /* Error handling */
+    }
+    ```
+
+2. Recognition result
+
+   When the hand gesture is recognized, the recognized result is delivered through `hand_gesture_recognition_cb()`.
+
+   You can [set the recognition result callback function using `hand_gesture_start_recognition()`](#start_stop).
```

Comment:
Question. In the "Start and stop recognizing Gesture" where linked to, `hand_gesture_recognition_cb()` is also described in sentence. So are we going to skip the code example for `hand_gesture_recognition_cb()`? Or we don't need the sample code?

Reply:
Actually, `hand_gesture_recognition_cb()` is set by `hand_gesture_start_recognition()`, and the sample code for setting `hand_gesture_recognition_cb()` is described in "Start and stop recognizing Gesture`. So, I linked "Start and stop recognizing Gesture" and didn't describe the sample code in this subsection. If you think that the sample code is necessary repeatedly, let me add the sample code. Thank you!!

Reply:
Thanks for the reply. I was simply comparing the description and the sample codes.
If so, can we consider the `recognition_cb()` in the sample code as `hand_gesture_recognition_cb()` in the description?

In that case we don't need to write the code repeatedly. How about just shortening the hyperlink like below?
Detailed information is proper, but too long hyperlink can be heavy.

   You can [set the recognition result callback function](#start_stop) using `hand_gesture_start_recognition()`.